### PR TITLE
[documentation] update code comments and contributing section

### PR DESCRIPTION
**Pull request by sarah-iot**

### CODE REVIEW

**Review by Vyrastas — APPROVED** [review]

File: include/openthread/history_tracker.h [hunk 1]
```diff
@@ -38,18 +38,12 @@ extern "C" {
...
  * @addtogroup api-history-tracker
  *
- * @{
- *
- */
-
-/**
  * @brief
- *   This header defines the public API for History Tracker.
+ *   Records the history of different events, for example RX and TX messages or network info changes. All tracked entries are timestamped.
```
Comment (jwhui) [inline include/openthread/history_tracker.h:42]: style: we limit line length to 120 characters
```suggestion
 *   Records the history of different events, for example RX and TX messages or network info changes. All tracked
 *   entries are timestamped.
```

**Comment by sarah-iot** [comment]
@jwhui OK, thanks...I made a few updates.
@Vyrastas

**Review by Vyrastas — APPROVED** [review]

**Comment by jwhui** [comment]
@sarah-iot , the pretty check and API version check are failing.
1. To resolve the API version check, we need to increment [`OPENTHREAD_API_VERSION`](https://github.com/openthread/openthread/blob/348f0a10a3dd735ac1e83e9df6209f7a2a61fd6a/include/openthread/instance.h#L56) when modifying any file in [include/](https://github.com/openthread/openthread/tree/main/include).
2. To resolve the pretty check, try running [`script/make-pretty`](https://github.com/openthread/openthread/blob/main/script/make-pretty).

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Comment by sarah-iot** [comment]
Thanks - I'll make sure to run those checks next time.

**Pull request merged**